test/index.spec.ts

```diff
@@ -0,0 +1,76 @@
+/// <reference types="@types/jest" />
+import { getSnapshot } from 'mobx-state-tree'
+
+import { persist } from '../src/index'
+import { UserStoreF, persistedDataF } from './fixtures'
+
+function getItem(key: string) {
+  const item = window.localStorage.getItem(key)
+  return item ? JSON.parse(item) : null // can only parse strings
+}
+
+describe('basic persist functionality', () => {
+  beforeEach(() => window.localStorage.clear())
+
+  it('should persist nothing if no actions are used', async () => {
+    const user = UserStoreF.create()
+    await persist('user', user)
+
+    expect(getItem('user')).toBe(null)
+  })
+
+  it('should persist snapshot when action used', async () => {
+    const user = UserStoreF.create()
+    await persist('user', user)
+
+    user.changeName('Joe') // fire action to trigger onSnapshot
+    expect(getItem('user')).toStrictEqual(getSnapshot(user))
+  })
+
+  it('should load persisted data', async () => {
+    window.localStorage.setItem('user', JSON.stringify(persistedDataF))
+
+    const user = UserStoreF.create()
+    await persist('user', user)
+    expect(getSnapshot(user)).toStrictEqual(persistedDataF)
+  })
+})
+
+describe('persist options', () => {
+  beforeEach(() => window.localStorage.clear())
+
+  it('shouldn\'t jsonify', async () => {
+    const user = UserStoreF.create()
+    await persist('user', user, {
+      jsonify: false
+    })
+
+    user.changeName('Joe') // fire action to trigger onSnapshot
+    // if not jsonified, localStorage will store as '[object Object]'
+    expect(window.localStorage.getItem('user')).toBe('[object Object]')
+  })
+
+  it('should whitelist', async () => {
+    const user = UserStoreF.create()
+    await persist('user', user, {
+      whitelist: ['name']
+    })
+
+    user.changeName('Joe') // fire action to trigger onSnapshot
+    const snapshot = { ...getSnapshot(user) }
```

Comment:
Might want to comment why this is necessary here and in the actual source code. I didn't know until now why it was necessary (as I had borrowed that piece from a gist). 

It throws a [`TypeError: Cannot delete property...`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Errors/Non_configurable_array_element) when trying to `delete` a key otherwise as the properties have apparently been set with `Object.defineProperty`.